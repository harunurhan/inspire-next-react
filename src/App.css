.menu {
  line-height: 64px;
}