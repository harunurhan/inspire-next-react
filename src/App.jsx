import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import { Layout, Menu } from 'antd';

import './App.css';
import Holdingpen from './holdingpen';


class App extends Component {
  render() {
    return (
      <div>
        <Layout>
          <Layout.Header>
            <Menu
              className="menu"
              theme="dark"
              mode="horizontal"
            >
              <Menu.Item>
                <Link to="/holdingpen">Holdingpen</Link>
              </Menu.Item>
            </Menu>
          </Layout.Header>

          <Layout.Content>
            <Route exact path="/holdingpen" component={Holdingpen} />
          </Layout.Content>
        </Layout>
      </div>
    );
  }
}

export default App;
