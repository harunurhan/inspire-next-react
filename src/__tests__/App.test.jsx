import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import App from '../App';
import Holdingpen from '../holdingpen';

describe('App', () => {
  it('renders initial state', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it('navigates to Holdingpen when /holdingpen', () => {
    const wrapper = mount((
      <MemoryRouter initialEntries={['/holdingpen']} initialIndex={0}>
        <App />
      </MemoryRouter>
    ));
    expect(wrapper.find(Holdingpen).length).toBe(1);
  });
});
