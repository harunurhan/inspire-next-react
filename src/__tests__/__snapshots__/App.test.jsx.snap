// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders initial state 1`] = `
<div>
  <Adapter>
    <Adapter>
      <Menu
        className="menu"
        mode="horizontal"
        prefixCls="ant-menu"
        theme="dark"
      >
        <MenuItem>
          <Link
            replace={false}
            to="/holdingpen"
          >
            Holdingpen
          </Link>
        </MenuItem>
      </Menu>
    </Adapter>
    <Adapter>
      <Route
        component={[Function]}
        exact={true}
        path="/holdingpen"
      />
    </Adapter>
  </Adapter>
</div>
`;
